tests/MongoDB.Driver.Examples/InsertDataWithEncryptedFieldsExample.cs

```diff
@@ -0,0 +1,243 @@
+                    {
+                        { "policyNumber", 123142 },
+                        { "provider", "MaestCare" }
+                    }
+                }
+            };
+
+        [Fact]
+        public void AutoEncryption()
+        {
+            var keyVaultCollectionNamespace = CollectionNamespace.FromFullName("encryption.__keyVault");
+            var recordsCollectionNamespace = CollectionNamespace.FromFullName("medicalRecords.patients");
+
+            var base64KeyId = ""; // paste the generated key in Standard GuidRepresentation form here
+
+            var ssnQuery = Builders<BsonDocument>.Filter.Eq("ssn", SampleSsnValue);
+
+            // Construct a JSON Schema
+            var schema = JsonSchemaCreator.CreateJsonSchema(base64KeyId);
+
+            // Construct an encrypted client
+            var encryptedClient = CreateEncryptedClient(
+                recordsCollectionNamespace,
+                keyVaultCollectionNamespace,
+                schema);
+            var collection = encryptedClient
+                .GetDatabase(recordsCollectionNamespace.DatabaseNamespace.DatabaseName)
+                .GetCollection<BsonDocument>(recordsCollectionNamespace.CollectionName);
+
+            // Insert a document into the collection
+            collection.UpdateOne(ssnQuery, new BsonDocument("$set", SampleDoc), new UpdateOptions() { IsUpsert = true });
+            Console.WriteLine("Successfully upserted the sample document!");
+
+            // Query SSN field with encrypted client
+            var result = collection.Find(ssnQuery).Single();
+
+            Console.WriteLine("Encrypted client query by the SSN (deterministically-encrypted) field:\n" + result.ToJson());
+
+            // Query SSN field with normal client without encryption
+            var normalMongoClient = new MongoClient(ConnectionString);
+            collection = normalMongoClient
+              .GetDatabase(recordsCollectionNamespace.DatabaseNamespace.DatabaseName)
+              .GetCollection<BsonDocument>(recordsCollectionNamespace.CollectionName);
+            var normalClientResult = collection.Find(ssnQuery).FirstOrDefault();
+            if (normalClientResult != null)
+            {
+                throw new Exception("Assert that the filtered data has not been found.");
+            }
+
+            // Query name (non-encrypted) field with normal client without encryption
+            var nameQuery = Builders<BsonDocument>.Filter.Eq("name", SampleNameValue);
+            var normalClientNameResult = collection.Find(nameQuery).FirstOrDefault();
+            if (normalClientNameResult == null)
+            {
+                throw new Exception("Assert that the filtered data has been found.");
+            }
+
+            Console.WriteLine($"Query by name returned the following document:\n {normalClientNameResult}.");
+        }
+
+        private IMongoClient CreateEncryptedClient(
+            CollectionNamespace recordNamespace,
```

Comment:
What does "record" mean? We don't normally use the word "record", we use "document".